Modules/Devices/src/Devices.Domain/Entities/Identities/Identity.cs

```diff
@@ -191,6 +192,28 @@ public void DeletionGracePeriodReminder3Sent()
         return DeletionProcesses.FirstOrDefault(x => x.Status == deletionProcessStatus);
     }
 
+    public static Expression<Func<Identity, bool>> IsWithAddress(IdentityAddress address)
+    {
+        return i => i.Address == address;
+    }
+
+    public static Expression<Func<Identity, bool>> IsReadyForDeletion()
+    {
+        return i => i.Status == IdentityStatus.ToBeDeleted && i.DeletionGracePeriodEndsAt != null && i.DeletionGracePeriodEndsAt < SystemTime.UtcNow;
+    }
+
+    public void DeletionStarted()
+    {
+        var deletionProcess = DeletionProcesses.SingleOrDefault(dp => dp.IsActive())
+                              ?? throw new DomainException(DomainErrors.NoActiveDeletionProcessFound());
+
+        if (deletionProcess.IsReadyToStartDeletion())
```

Comment:
This check is needed here because of the assignment in the following line. We could move it to the DeletionProperty domain class, but there is no navigation property to the Identity there. I'm not sure how we could work around this issue.